cibuildwheel/selector.py

```diff
@@ -87,7 +87,7 @@ def __call__(self, build_id: str) -> bool:
         # filter out groups that are not enabled
         if EnableGroup.CPythonFreeThreading not in self.enable and fnmatch(build_id, "cp313t-*"):
             return False
```

Comment:
_Not sure if there was any discussion about it already._ Should `cp314t` really be build by default? I'm not involved enough in the free threading ecosystem to know the current state of things. At least for mypy, I'd probably need to disable the builds manually to get the build pipeline to pass if the PR is merged as is.

The `enable = ["cpython-freethreading"]` option feels safer at least for the moment.

Reply:
Also see https://www.python.org/downloads/release/python-3140b3/

Reply:
Yes I did see that it's no longer considered experimental. Though I'd say that mainly applies for cpython itself. Not sure how far along the ecosystem itself is / how challenging it is to get the free threading builds to pass.

I'd suspect this will probably take another year or two, at least.

Reply:
Yes, but hopefully this will push people to start supporting it if they aren't already. The next stage will require most packages to support it, and I think the point of declaring it no longer experimental is to signal that you can't just ignore it anymore, which is also what we are also doing by requiring it to be explicitly skipped. Plus we explicitly said the enable was only going to be valid while it was experimental. :)

We used to require PyPy be explicitly skipped, and we still require musllinux to be explicitly skipped, so we have a bit of a history of requiring skips. :)

Reply:
Getting free threading working hasn't been that bad, though mostly people like @ngoldbaum and @colesbury have done a hero's share of the work, so maybe I shouldn't claim anything about how hard it was. ;) I'd guess it's not that bad for MyPyC, especially if you just supported 3.14t+, which is a little easier.

Reply:
We'd love to chat about adding support for free-threading in MyPyC.

I did the lion's share of the work for NumPy and PyO3 and our team has been involved in updating Cython, CFFI, SWIG and many other packages across the ecosystem. There's also the guide we're maintaining at https://py-free-threading.github.io.

All that to say we've accumulated a fair bit of know-how for this and we'd very much like to help out with MyPyC. In almost all cases we've looked at so far the issues have been more tractable than people initially feared.

Reply:
As I suspected at least for mypy there were quite a few build errors. This is just from a test run: https://github.com/cdce8p/mypy_mypyc-wheels/actions/runs/16496179019/job/46642337801#step:4:4390.

> We'd love to chat about adding support for free-threading in MyPyC.

This isn't my area of expertise per se. Feel free to engage on the issue tracker though. I believe this is the relevant one for it: https://github.com/mypyc/mypyc/issues/1038.

Reply:
To add one more datapoint only because I saw it just now: The dependabot update compatibility is at 33% currently. Obviously it's difficult to tell if that's because of 314, free threading or a combination of both.

Anyway the release notes make it quite obvious how to disable it so it isn't really an issue.